Lib/test/test_venv.py

```diff
@@ -360,6 +361,26 @@ def test_multiprocessing(self):
             'pool.terminate()'])
         self.assertEqual(out.strip(), "python".encode())
 
+    def test_deactivate_with_strict_bash_opts(self):
+        bash = shutil.which("bash")
+        if bash is None:
+            self.skipTest("bash required for this test")
+        rmtree(self.env_dir)
+        builder = venv.EnvBuilder(clear=True)
+        builder.create(self.env_dir)
+        activate = os.path.join(self.env_dir, self.bindir, 'activate')
+        test_script = os.path.join(self.env_dir, 'test_strict.sh')
+        with open(test_script, 'w') as f:
+            f.write("""
+set -euo pipefail
+source {}
+deactivate
+            """.format(activate))
+        out, err = check_output([bash, test_script])
```

Comment:
Turns out this fails on Windows on the Azure pipeline, because 'bash' is available there, despite being windows. To debug it I think this'll have to cach CalledProcessError and print the stderr.

Reply:
Alternatively you could just `@unittest.skipUnless(sys.platform == "win32")` that test or a better test.support equivalent API?

I _assume_ venv activation stuff isn't intended to work on bash on windows?  @zware as a random one to confirm that.

Reply:
I don't think there's really any reason for it not to.  The `activate` script is installed on Windows, and `source test_env/Scripts/activate` seems to work on my Windows 8.1 buildbot which has `bash` available at `C:\Program Files\Git\bin\bash.exe` (which is on PATH for the builder).

Reply:
I would guess that this failure on Azure is due to `bash` not actually being `bash`, but a redirection to "here's how to install bash".  @zooba might have a way to detect that?

Reply:
It gets pretty esoteric to check that specific of an external environment behavior reliably.  you basically need to test the output and exit code of `subprocess.run([bash_path, "-c", "echo 'this is actually bash' ; exit 99"])` or similar to see if your found bash acts like bash.  is it worth it?

Reply:
It's trying to run Git Bash, but there's no indication in the logged output what the error message is, so who knows?

I'm okay with skipping Bash activation on Windows.

Reply:
I went with skipping the test on Windows. I assume you would use activate.bat there anyways?